### PR TITLE
Update System.* deps to 4.5.0-rc1

### PR DESCRIPTION
**Pull request by iamcarbon**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Updates System.* deps to 4.5.0-rc1. there were no breaking changes. The API surface for System.Memory should finally be stable!

### CODE REVIEW

**Comment by iamcarbon** [comment]
Notable: All tests pass on .NETCORE 2.1.

File: tests/ImageSharp.Sandbox46/ImageSharp.Sandbox46.csproj [hunk 1]
```diff
@@ -33,7 +32,4 @@
   <ItemGroup>
     <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
-  <ItemGroup>
-    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
```
[the remark this answers is not in the document]
Comment (iamcarbon) [reply tests/ImageSharp.Sandbox46/ImageSharp.Sandbox46.csproj]: VS used to create these service references for test projects in the past. It hasn't for a few releases now.

[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply tests/ImageSharp.Sandbox46/ImageSharp.Sandbox46.csproj]: I don't think so as far as I'm aware.

**Comment by JimBobSquarePants** [comment]
Ooft! This is great! We can now finally expose the memory API's we need for interop and per row augmentation.

File: src/ImageSharp/ImageSharp.csproj [hunk 2]
```diff
@@ -41,8 +41,9 @@
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
     <PackageReference Include="System.Buffers" Version="4.4.0" />
-    <PackageReference Include="System.Memory" Version="4.5.0-preview2-26406-04" />
-    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.0-preview2-26406-04" />
+    <PackageReference Include="System.Memory" Version="4.5.0-rc1" />
+    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.0-rc1" />
+    <PackageReference Include="System.Numerics.Vectors" Version="4.5.0-rc1" />
```
Comment (JimBobSquarePants) [inline src/ImageSharp/ImageSharp.csproj]: Ideally we should update Core and let this dependency float through implicitly.

File: tests/ImageSharp.Sandbox46/ImageSharp.Sandbox46.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply tests/ImageSharp.Sandbox46/ImageSharp.Sandbox46.csproj]: @iamcarbon Good man. 👍

**Pull request merged**